opflexagent/firewall_wrapper.py

```diff
@@ -0,0 +1,181 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import contextlib
+from neutron.agent import firewall as fw
+from oslo_log import log as logging
+from oslo_utils import importutils
+
+
+LOG = logging.getLogger(__name__)
+
+
+class FirewallDriverWrapper(fw.FirewallDriver):
+    """Wrapper for Firewall Driver implemntations
+
+    This wrapper class keeps a map of strategies to
+    use for implementing a firewall driver. The strategy
+    selected is determined by a key, which is passed in
+    a map as part of the init_wrapper function. The strategy
```

Comment:
Nope -- I am :-). I'll change this to "set_strategy_and_map" (but only if we keep this class -- see additional comments).